### ISSUE TITLE
avahi-autoipd installs isc-dhcp-client hooks while ISC DHCP client is discontinued

### ISSUE THREAD

**Issue by mbiebl** [avahi/avahi#664]
`avahi-autoipd/dhclient-enter-hook.in  avahi-autoipd/dhclient-exit-hook.in` are hooks for ISC DHCP client which are installed during `make install`.
Given that ISC DHCP client has been discontinued maybe we should stop installing those integration files?

**Comment by mbiebl** [comment]
Begs the question, if there is a good alternative which avahi-autoipd should integrate with instead or if avahi-autoipd is still useful at all.

**Comment by evverx** [comment]
I think it's a duplicate of https://github.com/avahi/avahi/issues/467.

> if avahi-autoipd is still useful at all

To judge from https://github.com/avahi/avahi/labels/autoipd it's used in I assume legacy set-ups but these days there are maintained alternatives that can do the same thing.

**Comment by mbiebl** [comment]
hm yeah, it's pretty much a duplicate. That said, #467 asks for dhcpcd support in addition to isc-dhcp-client, while as this issues asks for the removal of the isc client hooks. So maybe still worth keeping open

**Comment by evverx** [comment]
> https://github.com/avahi/avahi/issues/467 asks for dhcpcd support in addition to isc-dhcp-client

Indeed. When I first read that issue I somehow assumed that it was about removing those hooks as well. Having taken a closer look it seems the removal wasn't mentioned there to probably smooth the switch.

>  maybe still worth keeping open

Makes sense. I think if autoipd was updated it would certainly make sense to remove those because there would be no reason to keep them upstream.

**Comment by evverx** [comment]
I've just closed #467 because `dhcpcd` supports IPv4LL out of the box so there is no need to hook it up to avahi-autoipd in the first place.

I saw https://salsa.debian.org/utopia-team/avahi/-/commit/1d196b153923c34a99aa4a923409424daf51404d. If you also have a patch that removes the hooks I think it should be OK to merge it to get it included in the next release.

**Comment by evverx** [comment]
Having thought about this personally I think it can safely be removed upstream: https://github.com/avahi/avahi/issues/467#issuecomment-2525416030
